src/app/g3w-ol/controls/screenshotcontrol.js

```diff
@@ -20,9 +20,27 @@ proto.change = function(layers=[]){
   this.setVisible(visible);
 };
 
+/**
+ * Check visibility map control based on layers
+ * @param layers <Array>
+ * @returns {boolean}
+ */
 proto.checkVisible = function(layers=[]){
-  const find = layers.find(layer => layer.isExternalWMS ? layer.isExternalWMS() : false);
-  return !find;
+  return "undefined" === typeof layers.find((layer) => {
+    if (layer.isExternalWMS && layer.isExternalWMS()) {
+      /**
+       * @since 3.8.0
+       * check if domain of wms is not that same of application to avoid CORS issue on getting
+       * map image
+       */
+      try {
+        const domain = new URL(layer.getSource().url);
+        // check if same origin of current page
+        return location.origin !== domain.origin;
+      } catch(err){}
+      return true;
+    } else return false
+  });
```

Comment:
I haven't tested, but to make it a little more immediate to the reader (and reusable elsewhere) I'd recommend rewriting it into something like the following:

```suggestion
/**
 * Check visibility map control based on layers
 *
 * @param {array} layers
 * @returns {boolean}
 */
proto.checkVisible = function(layers=[]) {
  return "undefined" === typeof layers.find((layer) => sameOrigin(location.href, layer.getSource().url));
};

/**
 * @TODO move this function into a shared "utility" file
 * @TODO handle try..catch for malformed URLs
 * 
 * Check for cross origin URLs.
 *
 * @since 3.8.0
 */
function sameOrigin(a, b) {
  return (new URL(a)).origin === (new URL(b)).origin;
}
```

Reply:
@Raruto we need to check is layer has source url otherwise sameOrigin method return an error because new URL(undefined) in not valid